frontend/src/cljs/imas_seamap/map/events.cljs

```diff
@@ -420,19 +419,22 @@
      :put-hash (encode-state db)
      :dispatch [:ui/hide-loading]}))
 
-(defn map-view-updated [{:keys [db]} [_ {:keys [zoom center bounds]}]]
+(defn map-view-updated [{:keys [db]} [_ {:keys [zoom size center bounds]}]]
   ;; Race-condition warning: this is called when the map changes to
   ;; keep state synchronised, but the map can start generating events
   ;; before the rest of the app is ready... avoid this by flagging
   ;; initialised state:
-  (when (:initialised db)
+  ;; Exceptions apply for size because we're only reading that from the map
```

Comment:
Does it matter that we set the size before it's initialised?

Reply:
Yes, it's necessary. If don't update it when the map is first created, then the next update won't happen until the user moves the map or resizes the window.
The map size is important because we need the size of the map to [calculate popup overflow](https://github.com/IMASau/Seamap/blob/f74698e30ba4281421e3f03a7ee4af0ba946c690/frontend/src/cljs/imas_seamap/map/events.cljs#L515-L517) and [how much to pan the map for each pixel of overflow](https://github.com/IMASau/Seamap/blob/f74698e30ba4281421e3f03a7ee4af0ba946c690/frontend/src/cljs/imas_seamap/map/events.cljs#L518-L519).
If the size of the map hasn't yet been set then we get some odd behaviour, like panning an infinite distance.

Reply:
Added comment to code for clarity.